services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_utils.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from contextlib import asynccontextmanager
+from typing import Any, AsyncGenerator, Dict
+
+import aiodocker
+from aiodocker.utils import clean_filters
+from fastapi import HTTPException, status
+
+logger = logging.getLogger(__name__)
+
+
+@asynccontextmanager
+async def docker_client() -> AsyncGenerator[aiodocker.Docker, None]:
+    docker = aiodocker.Docker()
+    try:
+        yield docker
+    except aiodocker.exceptions.DockerError as error:
+        logger.debug("An unexpected Docker error occurred", stack_info=True)
+        raise HTTPException(error.status, detail=error.message) from error
```

Comment:
why does this raise a HTTPException? in the core folder? Either move this to the api dependencies or else create a proper exception.

Reply:
Good point. Will just raise the DockerError here as it makes sense.